### PR TITLE
Add mappedBreakpoints

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

* cleans up breakpoint location logic
* will now select the exact location

### CODE REVIEW

**Comment by darkwing** [comment]
This makes me happy:

paused

**Comment by jasonLaster** [comment]
Yes!

**Comment by darkwing** [comment]
@jasonLaster I made a bunch of changes here so you may want to take another look at the general direction of this, flow issues aside

**Comment by jasonLaster** [comment]
sounds good

File: src/components/SecondaryPanes/Breakpoint.js [hunk 1]
```diff
@@ -38,21 +40,22 @@ function getBreakpointLocation(source, line, column) {
...
+  why: Why
+) {
+  if (!frame || isInterrupted(why)) {
+    return false;
   }
 
-  return originalText;
+  const bpId = getLocationWithoutColumn(breakpoint.location);
+  const pausedId = getLocationWithoutColumn(frame.location);
+  return bpId === pausedId;
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/Breakpoint.js]: maybe we just do the comparisons here

```js
bp.location.sourceId == frame.locaton.sourceId ...
```

**Comment by darkwing** [comment]
Weirdest way to find a bug ever but STR:

1.  Make a few breakpoints, refresh source document, wait for breakpoint to hit
2.  Close all tabs in the tab bar
Weirdest way to find a bug ever but STR:

1.  Make a few breakpoints, refresh source document, wait for breakpoint to hit
2.  Close all tabs in the tab bar

mappedbreakpointsbug

What's the best course of action if we suddenly don't have a `selectedSource`?  Returning or returning null?

**Comment by darkwing** [comment]
The error cited above is also persistent; i.e. if I try to refresh the debugger page the same error appears, so it seems we need to be super careful with storing breakpoint data.

**Comment by jasonLaster** [comment]
yep, returning early is good

File: src/selectors/mappedBreakpoints.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    location = breakpoint.generatedLocation || breakpoint.location;
+    text = breakpoint.text;
+  }
+  const localBP = { locationId, location, text, source, condition };
+
+  return localBP;
+}
+
+function _getMappedBreakpoints(breakpoints, sources, selectedSource) {
+  return breakpoints
```
Comment (jasonLaster) [inline src/selectors/mappedBreakpoints.js]: maybe check for the selectedSource here so that we can be extra cautious

Comment (darkwing) [reply src/selectors/mappedBreakpoints.js]: Returning a `I.Map()` seems like the best idea, IMO, because it can prevent breakage elsewhere.

**Comment by darkwing** [comment]
I'm also seeing this periodically; trying to figure out if it's an existing issue or new to this PR:

mapscopeserror

**Comment by jasonLaster** [comment]
it would be great to do this https://github.com/devtools-html/debugger.html/issues/5875 after this lands because then we wont need shouldComponentUpdate

**Pull request merged**